### PR TITLE
Unit 2 james franks

### PR DESCRIPTION
**Pull request by jayfr1122**
# Submitting Unit 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ x] Coding Challenges
- [ x] Coding Concepts
- [ x] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -33,11 +53,31 @@ const object2 = { number: 0 }
...
+    it("takes a object as an argument and decides if the number inside it is evenly divisible by three or not.", () => {
+        const object1 = { number: 15 }
+        const object2 = { number: 0 }
+        const object3 = { number: -7 }
+        expect(divisByThree(object1)).toEqual("15 is divisible by three")
+        expect(divisByThree(object2)).toEqual("0 is divisible by three")
+        expect(divisByThree(object3)).toEqual("-7 is not divisible by three")
+    })
+})
+//Fails at line 61 expect.
```
Comment (sjproctor) [inline code-challenges.test.js:66]: Your test looks great. Since you have the test variables in the test (as they should be) they shouldn't be duplicated outside the test.

File: code-challenges.test.js [hunk 2]
```diff
@@ -33,11 +53,31 @@ const object2 = { number: 0 }
...
+    return `${value.number} is not divisible by three`
+    }
+}
+//Passes the test.
 // Pseudo code:
-
+// -create test
+// -takes an object as an argument 
+// -decides if the number is evenly divisible by 3 or not
+// -create function
```
Comment (sjproctor) [inline code-challenges.test.js:80]: Typically the "create function" part is the focus of the pseudo code. What the function does is the part that matters.

File: code-challenges.test.js [hunk 3]
```diff
@@ -46,7 +86,27 @@ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
...
\ No newline at end of file
+const capWord = (array) => {
+    return array.map(value => {
+        if (typeof value === "string" && value.length > 0) {
+            return value[0].toUpperCase() + value.slice(1)
+        }else {
+            return value
+        }
+    })
+}
```
Comment (sjproctor) [inline code-challenges.test.js:107]: Interesting edge case handling with the typeof and length check.

File: coding-concepts.js [hunk 4]
```diff
@@ -9,28 +9,28 @@
...
 const greeter = (name) => {
   `Hello, ${name}!`
 }
-// console.log(greeter("LEARN Student"))
+console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: It will not return the intended `Hello, LEARN Student!`
+// b) Verify and explain: The computer returned undefined. The reason the computer returned undefined is because the code lacked the return before the string interpolation.
```
Comment (sjproctor) [inline coding-concepts.js:25]: Nice!

**Review by sjproctor — APPROVED** [review]
Nice work. Your coding concepts and interview sections look pretty good. Your test structure looks great just make sure your test variables are not duplicated. Your focus area is your pseudo code. This is a competent skill level assessment.
